utils/rsaencryption/rsa_encryption_test.go

```diff
@@ -46,3 +55,56 @@ func TestPrivateKeyToByte(t *testing.T) {
 	require.NotNil(t, b)
 	require.Greater(t, len(b), 1024)
 }
+
+func TestConvertEncryptedPemToPrivateKey(t *testing.T) {
+	keystorePassword := "123123123"
+	generatedPrivateKey, err := rsa.GenerateKey(rand.Reader, keySize)
+	require.NoError(t, err)
+	privDER := x509.MarshalPKCS1PrivateKey(generatedPrivateKey)
+
+	// Create a pem.Block with the private key.
+	privateBlock := pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privDER,
+	}
+
+	// Encode the pem.Block with the private key into a temporary file.
+	tmpPrivateFile, err := os.CreateTemp("", "private_key.pem")
+	require.NoError(t, err)
+	err = pem.Encode(tmpPrivateFile, &privateBlock)
+	require.NoError(t, err)
+	err = tmpPrivateFile.Close()
+	require.NoError(t, err)
+
+	// Encrypt the private key file using OpenSSL.
+	passString := "pass:" + keystorePassword
+	cmd := exec.Command("openssl", "rsa", "-aes256", "-in", tmpPrivateFile.Name(), "-out", "encrypted_private_key.pem", "-passout", passString)
+	err = cmd.Run()
+	require.NoError(t, err)
+
+	// Delete the temporary unencrypted private key file.
+	err = os.Remove(tmpPrivateFile.Name())
+	require.NoError(t, err)
+
+	// Read the encrypted private key file.
+	pemBytes, err := os.ReadFile("encrypted_private_key.pem")
+	require.NoError(t, err)
+	hash := sha256.Sum256(pemBytes)
+	fmt.Printf("SHA-256: %s\n", hex.EncodeToString(hash[:]))
+
+	// Convert encrypted PEM to private key.
+	privateKey, err := ConvertEncryptedPemToPrivateKey(pemBytes, keystorePassword)
+	require.NoError(t, err)
+	require.Equal(t, privateKey, generatedPrivateKey)
+
+	// Test with incorrect password.
+	_, err = ConvertEncryptedPemToPrivateKey(pemBytes, keystorePassword+"1")
+	require.Error(t, err)
+
+	_, err = ConvertEncryptedPemToPrivateKey(pemBytes, "")
+	require.Error(t, err)
+
+	// Clean up the encrypted private key file.
+	err = os.Remove("encrypted_private_key.pem")
```

Comment:
I think the file needs to be removed in `defer` statement after creation, because `require`s above may interrupt the test execution and the file would not be deleted